schemas/stsci.edu/asdf/0.1.0/core/asdf.yaml

```diff
@@ -11,6 +11,20 @@ description: |
 tag: "tag:stsci.edu:asdf/0.1.0/core/asdf"
 type: object
 properties:
+  asdf_library:
+    description: |
+      Describes the ASDF library that produced the file.
+    $ref: software
+
+  software:
+    description: |
+      Lists any relevant software that produced or modified the ASDF
+      file.  May include such things as data calibration pipelines,
+      data analysis tools etc.
+    type: array
+    items:
+      $ref: "software"
```

Comment:
What if we instead turned this into a full "history" log, including the software that made the change, the timestamp, an an optional log message (which would be passed in by the user via the API when saving or updating a file).
Could include other optional log metadata but I'm not sure how crazy to go with this.